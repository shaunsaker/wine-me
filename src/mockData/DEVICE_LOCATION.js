const DEVICE_LOCATION = {
  latitude: -34.9263749,
  longitude: 18.9319848,
};

export default DEVICE_LOCATION;
