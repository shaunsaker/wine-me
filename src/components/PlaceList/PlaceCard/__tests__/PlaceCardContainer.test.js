import React from 'react';
import renderer from 'react-test-renderer';

import { PlaceCardContainer } from '..';
import PLACE from '../../../../mockData/PLACE';
import DEVICE_LOCATION from '../../../../mockData/DEVICE_LOCATION';

describe('PlaceCard', () => {
  const spies = [];

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(
        <PlaceCardContainer place={PLACE} deviceLocation={DEVICE_LOCATION} />,
      );

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle onPress', () => {
      spies[0] = jest.spyOn(PlaceCardContainer.prototype, 'navigate');
      const component = renderer.create(
        <PlaceCardContainer place={PLACE} deviceLocation={DEVICE_LOCATION} />,
      );
      const instance = component.getInstance();

      instance.onPress();

      expect(spies[0]).toHaveBeenCalledWith('place', { place: PLACE });
    });

    it('should handle navigate', () => {
      const component = renderer.create(
        <PlaceCardContainer place={PLACE} deviceLocation={DEVICE_LOCATION} />,
      );
      const instance = component.getInstance();

      instance.navigate();
    });
  });

  describe('actions', () => {
    it('should call onPress on press', () => {
      spies[0] = jest.spyOn(PlaceCardContainer.prototype, 'onPress');
      const component = renderer.create(
        <PlaceCardContainer place={PLACE} deviceLocation={DEVICE_LOCATION} />,
      );
      const { root } = component;
      const targetComponent = root.findByProps({ testID: 'placeCardContainer' });

      targetComponent.props.handlePress();

      expect(spies[0]).toHaveBeenCalled();
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
  });
});
