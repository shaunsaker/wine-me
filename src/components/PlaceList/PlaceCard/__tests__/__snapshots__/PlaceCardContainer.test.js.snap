// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard renders renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignSelf": "stretch",
      "backgroundColor": "#FFFFFF",
      "borderColor": null,
      "borderRadius": 5,
      "borderWidth": 0,
      "elevation": 2,
      "opacity": 1,
      "overflow": "hidden",
      "shadowColor": "#212121",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.17,
      "shadowRadius": 2,
    }
  }
  testID="placeCardContainer"
>
  <View
    style={Object {}}
  >
    <View
      style={
        Array [
          Array [
            Object {
              "flex": 1,
              "height": 200,
            },
            undefined,
          ],
          Object {
            "overflow": "hidden",
          },
        ]
      }
    >
      <FastImageView
        borderRadius={5}
        onFastImageError={[Function]}
        onFastImageLoadEnd={[Function]}
        onFastImageLoadStart={[Function]}
        resizeMode="cover"
        source={
          Object {
            "uri": "https://maps.googleapis.com/maps/api/place/photo?maxheight=200&photoreference=CmRaAAAAF5bkjtj7bFEGaZqTN1_uZrJMvM6EEOFle3UIDd5JEbZwb281f2F4EH-XR1o6frihMiOzrJEx1KlczmV8o4qLrYygDZE-TKthSRT1SdzvrEQ6AjQbCG6sVeZBR48V0lMLEhDkpyqJh9O9z4Ai-7All7swGhQwTkHxJ77yIZ9d0c8oqWifUrL6eA&key=AIzaSyAu5M0b3zBi4gT50BjpspZnEtCNeyCJwvY",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <BVLinearGradient
    colors={
      Array [
        0,
        3003121664,
      ]
    }
    endPoint={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "flex-start",
        "borderBottomLeftRadius": 5,
        "borderBottomRightRadius": 5,
        "bottom": 0,
        "left": 0,
        "paddingHorizontal": 15,
        "paddingTop": 30,
        "paddingVertical": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Nunito-SemiBold",
          "fontSize": 17,
          "marginBottom": 10,
        }
      }
    >
      Bartinney Tasting Shed
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#E23E57",
          "borderColor": null,
          "borderRadius": 5,
          "borderWidth": 0,
          "elevation": 2,
          "opacity": 1,
          "paddingHorizontal": 7.5,
          "paddingVertical": 5,
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.17,
          "shadowRadius": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Nunito-SemiBold",
            "fontSize": 11,
          }
        }
      >
        111 km from you
      </Text>
    </View>
  </BVLinearGradient>
  <View
    style={
      Object {
        "position": "absolute",
        "right": 10,
        "top": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderColor": null,
          "borderRadius": 15,
          "borderWidth": 0,
          "elevation": 2,
          "height": 30,
          "justifyContent": "center",
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.17,
          "shadowRadius": 2,
          "width": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#4caf50",
              "fontSize": 24,
            },
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        î—Š
      </Text>
    </View>
  </View>
</View>
`;
